File: client/lib/models/Activity.dart
// ignore: duplicate_ignore
// ignore_for_file: file_names, duplicate_ignore, empty_constructor_bodies
// ignore: file_names
enum Edifficulty {hard, medium, easy}
enum Eseason {summer, spring, winter, fall}

class Activity {
  final int id;
  final String name;
  final String picture;
  final String review;
  final int rating;
  final int durationinHours;
  final Edifficulty difficulty;
  final Eseason season;

  Activity (this.id, this.name, this.picture, this.review, this.rating, this.durationinHours, this.difficulty, this.season);

  
  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'name': name,
      'picture': picture,
      'review': review,
      'rating': rating,
      'durationinHours': durationinHours,
      'difficulty': difficulty.toString().split('.').last,
      'season': season.toString().split('.').last,
    };
  }
}

File: api/src/models/Activity.js
/**Activity Model:
ID, 
Nombre, 
Imagen (URL), 
Dificultad, 
Calificación, 
Duración en horas, 
Época.
*/
const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
    sequelize.define('Activity', {
        ID: {
            type: DataTypes.INTEGER,
            primaryKey: true,
            allowNull: false,
            autoIncrement: true
        },
        name: {
            type: DataTypes.STRING,
            allowNull: false
        },
        picture: {
            type:DataTypes.TEXT,
            allowNull:false
        },
        review: {
            type:DataTypes.TEXT,
        },
        difficulty: {
            type: DataTypes.ENUM('Hard','Medium', 'Easy'),
            allowNull: false 
        },
        rating: {
            type: DataTypes.SMALLINT,
            allowNull: false,
            validate: {
                min: 1,
                max: 5
            }
        },
        duration_inHours: {
            type: DataTypes.SMALLINT,
        },
        season: {
            type: DataTypes.ENUM('Summer','Spring', 'Winter', 'Fall'),
            allowNull: false
        }
    });
};
